### PR TITLE
Properly sets hands to use held_index so the code knows they're hands.

### LINKED ISSUE #20457: Prosthetic and Augmented Replacement Arms Don't Work
A recent round involved a lot of places around the station getting bombed. As one of the victims, my character got a limb blown off. However, even after the arm was reattached, it was nonfunctional. I tried getting it augmented - no change. As a doctor, I helped other people who'd been caught in the mayhem - replacing a janitor's arm straight away with a cyborg limb led to him having the same problem.

Clearly, arm replacements should work - otherwise, what's the point of having them? Nothing else extraordinary happened - getting my arm blown off was one of the first parts of the round, it was with a model ship bomb. Picked up the arm from the breach, went to get it put back on.

Was playing on Sybil, Box was the Map. Server revision "ce298aa381eea52140b67f21ac27605695153dcf" with test merges of "#20441" and "#20445".

### PR DESCRIPTION
**Pull request by Incoming5643**
Fixes #20457 

The code was fine, it just was never ran for limbs spawned outside a human because this was forgotten, or lost in a merge resolution, or something.

Should not of taken three weeks guys.

### CODE REVIEW

**Comment by lzimann** [comment]
Wouldnt this be a problem with multiple hands? Nain reason on why I didn't fix it myself was because I had no idea how to do it support multiple hands...

**Comment by Incoming5643** [comment]
I'm not clear how/if this works with multiple hands, I just know that this clearly wasn't set anywhere when the code intended it to be. Frankly those are secondary problems until a multi armed race is actually a thing that exists.

**Comment by RemieRichards** [comment]
They were set in ```human/New()```, and later moved to ```create_bodyparts()``` by Phil.
the definition never got values because idk why.

Since you're touching indexes, I'm gonna need you to change ```create_bodyparts()``` to this:

```
/mob/living/carbon/proc/create_bodyparts()
        var/l_arm_index_next = -1
        var/r_arm_index_next = 0
	for(var/X in bodyparts)
		var/obj/item/bodypart/O = new X()
		O.owner = src
		bodyparts.Remove(X)
		bodyparts.Add(O)
		if(O.body_part == ARM_LEFT)
			O.held_index = (l_arm_index_next += 2) //1, 3, 5, 7...
			hand_bodyparts += O
		else if(O.body_part == ARM_RIGHT)
			O.held_index = (r_arm_index_next += 2) //2, 4, 6, 8...
			hand_bodyparts += O
```

so if a species has multiple arms they get proper indexes.

**Comment by Incoming5643** [comment]
Oh well, if that's the case this doesn't fix anything.

**Comment by RemieRichards** [comment]
@Incoming5643 it fixes spawned limbs, which is decent enough I suppose.
Since I made our version of multihands I really did intend for l/r arms to have *SOME* kind of held index value, I must have just forgotten it in the def.

**Comment by Incoming5643** [comment]
done and done

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -725,16 +725,18 @@
...
+	var/l_arm_index_next = -1
+	var/r_arm_index_next = 0
 	for(var/X in bodyparts)
 		var/obj/item/bodypart/O = new X()
 		O.owner = src
 		bodyparts.Remove(X)
 		bodyparts.Add(O)
 		if(O.body_part == ARM_LEFT)
-			O.held_index = 1
+			O.held_index = (l_arm_index_next += 2) //1, 3, 5, 7...
```
Comment (phil235) [inline code/modules/mob/living/carbon/carbon.dm]: `O.held_index = l_arm_index_next + 2`
same for below

Comment (RemieRichards) [inline code/modules/mob/living/carbon/carbon.dm]: No phil, reread the logic.

Comment (Incoming5643) [inline code/modules/mob/living/carbon/carbon.dm]: In any case it doesn't compile like this

Comment (RemieRichards) [inline code/modules/mob/living/carbon/carbon.dm]: Yes it does compile, I tested it locally before I even suggested the idea.
511 upwards, which we should be targeting by now (heck, i think it's already on the servers, @MrStonedOne ?)

Comment (Incoming5643) [inline code/modules/mob/living/carbon/carbon.dm]: It's on the servers but not required of the clients. Stable is still in 510.

Comment (RemieRichards) [inline code/modules/mob/living/carbon/carbon.dm]: Yes @Incoming5643, and this is a server side change you buttface.

Comment (Incoming5643) [inline code/modules/mob/living/carbon/carbon.dm]: I just wan travis to be happy no bully

**Pull request merged**